src/ngraph/runtime/gpu/pass/gpu_rnn_fusion.cpp

```diff
@@ -0,0 +1,848 @@
+        if (input_xt_rank != 2 || hidden_ht_rank != 2 || weights_i2h_rank != 2 ||
+            weights_h2h_rank != 2)
+        {
+            return false;
+        }
+
+        if (bias_i2h->get_shape().size() != 1 || bias_h2h->get_shape().size() != 1)
+        {
+            throw ngraph_error("Bias should have rank of 1 for Rnn op");
+        }
+
+        // Determine which is ht_1 and xt. but if both xt and ht_1 have the same shape we need to capture this
+        // reliably in the RNN fusion.
+        std::shared_ptr<op::gpu::Lstm> lstm = nullptr;
+        bool intermediate_lstm = false;
+
+        if (std::dynamic_pointer_cast<op::GetOutputElement>(pattern_map[ct_1]))
+        {
+            intermediate_lstm = true;
+        }
+
+        // if the matched LSTM is the first cell we need to check if symbol input_xt corresponds
+        // to the input data tensor, or the hidden (recurrent) data tensor
+        if (!intermediate_lstm &&
+            (std::dynamic_pointer_cast<op::Broadcast>(pattern_map[hidden_ht]) &&
+             std::dynamic_pointer_cast<op::Constant>(pattern_map[hidden_ht]->get_argument(0))))
+        // label input_xt is the input data to the first LSTM
+        {
+            lstm = std::make_shared<op::gpu::Lstm>(pattern_map[input_xt],
+                                                   pattern_map[weights_i2h],
+                                                   pattern_map[hidden_ht],
+                                                   pattern_map[weights_h2h],
+                                                   pattern_map[bias_i2h],
+                                                   pattern_map[bias_h2h],
+                                                   pattern_map[ct_1]);
+        }
+        else if (!intermediate_lstm &&
+                 (std::dynamic_pointer_cast<op::Broadcast>(pattern_map[input_xt]) &&
+                  std::dynamic_pointer_cast<op::Constant>(pattern_map[input_xt]->get_argument(0))))
+        // label hidden_ht is the input data to the first LSTM
+        {
+            lstm = std::make_shared<op::gpu::Lstm>(pattern_map[hidden_ht],
+                                                   pattern_map[weights_h2h],
+                                                   pattern_map[input_xt],
+                                                   pattern_map[weights_i2h],
+                                                   pattern_map[bias_h2h],
+                                                   pattern_map[bias_i2h],
+                                                   pattern_map[ct_1]);
+        }
+        else if (pattern_map[hidden_ht]->get_arguments().size() &&
+                 pattern_map[ct_1]->get_arguments().at(0)->get_instance_id() ==
+                     pattern_map[hidden_ht]->get_arguments().at(0)->get_instance_id())
+        // this still has a bug vector: if the hidden input ht is a non-broadcasted constant
+        // it will be misclassified as input data xt
```

Comment:
can we have case, where it will be not broadcasted from constant?, usually the layout will be `tnc` or `ntc` . so i guess`constant` will be end up getting broadcasted.